pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -615,7 +618,47 @@ private static Expression toExpression(SqlNode node) {
             funcExpr.getFunctionCall().addToOperands(toExpression(child));
           }
         }
+
+        if (FunctionDefinitionRegistry.isScalarFunc(funcName) && isCompileTimeEvaluationPossible(funcExpr)) {
+
+          ScalarFunctionType scalarFunctionType = ScalarFunctionType.getScalarFunctionType(funcName);
+          switch (scalarFunctionType) {
+            case NOW:
```

Comment:
Will this approach scale, as number of scalar functions increase? For example, each one would need to be added here. What do you think about modelling this as a query rewrite phase that goes over all scalars and evaluates them?

Reply:
I think we will move to function registry/invoker model

Reply:
yes, this is to just ensure that the logic of detection if a function can be evaluated at query compile time works.